atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/TableMetadata.java

```diff
@@ -185,149 +110,64 @@ public LogSafety getNameLogSafety() {
+        builder.setNameLogSafety(getNameLogSafety());
         return builder;
     }
 
     public static TableMetadata hydrateFromProto(TableMetadataPersistence.TableMetadata message) {
-        CachePriority cachePriority = CachePriority.WARM;
+        ImmutableTableMetadata.Builder builder = builder()
+                .rowMetadata(NameMetadataDescription.hydrateFromProto(message.getRowName()))
+                .columns(ColumnMetadataDescription.hydrateFromProto(message.getColumns()))
+                .conflictHandler(ConflictHandlers.hydrateFromProto(message.getConflictHandler()));
+
         if (message.hasCachePriority()) {
-            cachePriority = message.getCachePriority();
+            builder.cachePriority(message.getCachePriority());
         }
-        boolean rangeScanAllowed = false;
         if (message.hasRangeScanAllowed()) {
-            rangeScanAllowed = message.getRangeScanAllowed();
+            builder.rangeScanAllowed(message.getRangeScanAllowed());
         }
-        int explicitCompressionBlockSizeKB = 0;
         if (message.hasExplicitCompressionBlockSizeKiloBytes()) {
-            explicitCompressionBlockSizeKB = message.getExplicitCompressionBlockSizeKiloBytes();
+            builder.explicitCompressionBlockSizeKB(message.getExplicitCompressionBlockSizeKiloBytes());
         }
-        boolean negativeLookups = false;
         if (message.hasNegativeLookups()) {
-            negativeLookups = message.getNegativeLookups();
+            builder.negativeLookups(message.getNegativeLookups());
         }
-        SweepStrategy sweepStrategy = SweepStrategy.CONSERVATIVE;
         if (message.hasSweepStrategy()) {
-            sweepStrategy = message.getSweepStrategy();
+            builder.sweepStrategy(message.getSweepStrategy());
         }
-        boolean appendHeavyAndReadLight = false;
         if (message.hasAppendHeavyAndReadLight()) {
-            appendHeavyAndReadLight = message.getAppendHeavyAndReadLight();
+            builder.appendHeavyAndReadLight(message.getAppendHeavyAndReadLight());
         }
-        LogSafety nameLogSafety = LogSafety.UNSAFE;
         if (message.hasNameLogSafety()) {
-            nameLogSafety = message.getNameLogSafety();
+            builder.nameLogSafety(message.getNameLogSafety());
         }
 
-        return new TableMetadata(
-                NameMetadataDescription.hydrateFromProto(message.getRowName()),
-                ColumnMetadataDescription.hydrateFromProto(message.getColumns()),
-                ConflictHandlers.hydrateFromProto(message.getConflictHandler()),
-                cachePriority,
-                rangeScanAllowed,
-                explicitCompressionBlockSizeKB,
-                negativeLookups,
-                sweepStrategy,
-                appendHeavyAndReadLight,
-                nameLogSafety);
+        return builder.build();
     }
 
     @Override
     public String toString() {
```

Comment:
you should be able to delete this?

Reply:
I think the generated one is slightly different, but we should be able to live with that userbreak